pymchelper/utils/mcscripter.py

```diff
@@ -0,0 +1,287 @@
+import sys
+import logging
+
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+
+class Config():
+    """
+    Reading the config file.
+    """
+    def __init__(self, fn):
+        with open(fn) as file:
+            self.lines = file.readlines()
+            self.base_dir = os.path.dirname(fn)
+
+            # script must run relative to location of config file.
+            if self.base_dir:
+                os.chdir(self.base_dir)
+            self.parse()
+
+    def parse(self):
+        """
+        Parse configuration file.
+        All data are read into dicts. There are two dicts: a constant and variable (tabluated) one.
+        """
+
+        self.c_dict = {}  # list of contant assignments
+        self.t_dict = {}  # list of table assignments
+
+        keys = []
+        vals = []
+
+        _first_line_in_table = True
+
+        for line in self.lines:
+            # skip comments
+            if line[0] == '#':
+                continue
+
+            # constant assigments
+            if "=" in line:
+                _v = line.split("=")
+                self.c_dict[_v[0].strip()] = _v[1].strip()
+                continue
+
+            # check if we are starting a table
+            if len(line.split()) > 0:
+                if _first_line_in_table:
+                    keys = line.split()
+                    _first_line_in_table = False
+                else:
+                    vals.append(line.split())
+
+        # after parsing all lines setup the variable table, if it exists:
+        if keys and vals:
+            for i, key in enumerate(keys):
+                self.t_dict[key] = [val[i] for val in vals]
+
+        _f = []
+        for item in self.c_dict["FILES"].split(","):
+            _f.append(item.strip())
+        self.c_dict["FILES"] = _f
+
+        _f = []
+        for item in self.c_dict["SYMLINKS"].split(","):
+            _f.append(item.strip())
+        self.c_dict["SYMLINKS"] = _f
+
+
+class McFile():
+    """
+    General MC single file object.
+    This will be used for the template files as well as the generated output files.
+    """
+    def __init__(self):
+        self.fname = ""  # filename
+        self.path = ""   # full path
+        self.lines = []  # list of lines inside this file
+        self.symlink = False  # marker if file is a symlink
+        self.tdir = ""
+
+    def write(self):
+        """
+        Write self to disk, create symlink if that is the case.
+        """
+        # check if target directory exists, create it, if not.
+        try:
+            os.makedirs(os.path.dirname(self.path))
+        except FileExistsError:
+            pass
+
+        if self.symlink:
+            link_file = os.path.join(self.tdir, self.fname)
+            link_target = os.path.join(self.path)
+            link_name = os.path.relpath(link_file, os.path.dirname(link_target))
+
+            try:
+                os.symlink(link_name, link_target)
+            except FileExistsError:
```

Comment:
It seems python2 doesn't support such Exception

Reply:
Better to use something low-level like:
```
import os
import errno

try:
    os.mkdir(folderPath)
except OSError as e:
    if e.errno == errno.EEXIST:
        print('Directory not created.')
    else:
        raise
```

from https://stackoverflow.com/a/20790635